### PR TITLE
fix: missing file error and swallowing error

### LINKED ISSUE #10581: When a failure occurs in Amplify the diagnose workflow swallows the error
### Before opening, please confirm:

- [X] I have installed the latest version of the Amplify CLI (see above), and confirmed that the issue still persists.
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-cli/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#bug-reports).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### How did you install the Amplify CLI?

npm

### If applicable, what version of Node.js are you using?

node 14

### Amplify CLI Version

8.5.0

### What operating system are you using?

linux

### Did you make any manual changes to the cloud resources managed by Amplify? Please describe the changes made.

No

### Amplify Categories

Not applicable

### Amplify Commands

Not applicable

### Describe the bug

When a failure occurs on the Amplify CLI the diagnose command gets invoked even when it is not project specific command like Amplify upgrade.

### Expected behavior

Shouldn't swallow the error 

### Reproduction steps

amplify init -y
delete the `backend-config.json` file
ampfliy diagnose --send-report
and see the error
```

Learn more at https://docs.amplify.aws/cli/reference/diagnose/
⠋ Creating Zip🛑 File at path: '/Users/ammkara/debugdiagnose/amplify/backend/backend-config.json' does not exist
Error: File at path: '/Users/ammkara/debugdiagnose/amplify/backend/backend-config.json' does not exist
    at Function.readJson (/snapshot/repo/build/node_modules/amplify-cli-core/lib/jsonUtilities.js:40:19)
    at StateManager.getData (/snapshot/repo/build/node_modules/amplify-cli-core/lib/state-manager/stateManager.js:309:56)
    at StateManager.getBackendConfig (/snapshot/repo/build/node_modules/amplify-cli-core/lib/state-manager/stateManager.js:112:25)
    at createZip (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/commands/diagnose.js:128:53)
    at zipSend (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/commands/diagnose.js:99:35)
    at reportError (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/commands/diagnose.js:62:15)
    at Object.run (/snapshot/repo/build/node_modules/@aws-amplify/cli-internal/lib/index.js:168:46)
    at processTicksAndRejections (internal/process/task_queues.js:95:5)
```

### GraphQL schema(s)

<details>

```graphql
# Put schemas below this line


```

</details>

### Log output

<details>

```
# Put your logs below this line


```

</details>

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by ammarkarachi**
#### Description of changes

fixes #10581, #10582
#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [x] Tests are [changed or added](https://github.com/aws-amplify/amplify-cli/blob/master/CONTRIBUTING.md#tests)
- [x] Relevant documentation is changed or added (and PR referenced)
- [x] New AWS SDK calls or CloudFormation actions have been added to relevant test and service IAM policies

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-cli/src/commands/helpers/collect-files.ts [hunk 1]
```diff
@@ -61,22 +61,32 @@ export const collectFiles = (resources: { category: string, resourceName: string
...
+  }
 
   // backend config file
-  filePaths.push({
-    filePath: pathManager.getBackendConfigFilePath(rootPath),
-    redact: false,
-  });
+  const backendConfigFile = pathManager.getBackendConfigFilePath(rootPath)
+  if(fs.existsSync(backendConfigFile)) {
+    console.log(backendConfigFile);
```
Comment (jhockett) [inline packages/amplify-cli/src/commands/helpers/collect-files.ts:84]: leftover?

File: packages/amplify-cli/src/commands/diagnose.ts [hunk 2]
```diff
@@ -1,23 +1,21 @@
...
-import { v4 } from 'uuid';
-import { prompter, printer } from 'amplify-prompts';
-import { collectFiles } from './helpers/collect-files';
-import { encryptBuffer, encryptKey, createHashedIdentifier } from './helpers/encryption-helpers';
-import { UsageDataPayload } from '../domain/amplify-usageData/UsageDataPayload';
-import { DebugConfig } from '../app-config/debug-config';
-import { isHeadlessCommand } from '../utils/headless-input-utils';
-import { Context } from '../domain/context';
-import { reporterEndpoint } from './helpers/reporter-apis';
+import { stateManager, pathManager, NotInitializedError, spinner, DiagnoseReportUploadError } from "amplify-cli-core";
```
Comment (jhockett) [inline packages/amplify-cli/src/commands/diagnose.ts:1]: The eslint config should prefer single quotes, do you have a vscode setting that caused this?

Comment (ammarkarachi) [reply packages/amplify-cli/src/commands/diagnose.ts:1]: I am not sure, my linter keeps crashing on my machine

File: packages/amplify-cli/src/commands/helpers/collect-files.ts (hunk 1 above)
Comment (ammarkarachi) [reply packages/amplify-cli/src/commands/helpers/collect-files.ts:84]: yeah 🙈

Comment (ammarkarachi) [reply packages/amplify-cli/src/commands/helpers/collect-files.ts]: fixed

File: packages/amplify-cli/src/commands/diagnose.ts (hunk 2 above)
Comment (ammarkarachi) [reply packages/amplify-cli/src/commands/diagnose.ts]: fixed

File: packages/amplify-cli/src/commands/diagnose.ts [hunk 3]
```diff
@@ -89,28 +92,30 @@ const zipSend = async (context: Context, skipPrompts: boolean, error: Error | un
...
       printer.blankLine();
       printer.info(`Project Identifier: ${projectId}`);
       printer.blankLine();
-    } catch (ex) {
-      context.usageData.emitError(ex);
-      spinner.fail();
     }
+  } catch (ex) {
+    printer.blankLine()
+    console.log(ex.message)
```
Comment (jhockett) [inline packages/amplify-cli/src/commands/diagnose.ts:116]:
```suggestion
    printer.info(ex.message)
```

**Review by jhockett — APPROVED** [review]

**Comment by danielleadams** [comment]
@ammarkarachi is there tests that you can write for this?

**Comment by ammarkarachi** [comment]
@danielleadams Done

**Review by danielleadams — APPROVED** [review]

**Pull request merged**